qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -126,310 +145,192 @@ def result(self, timeout=None, wait=5):
-        Returns whether job is queued.
-
-        Returns:
-            bool: True if job is queued, else False.
-
-        Raises:
-            QISKitError: couldn't get job status from server
-        """
-        return self.status['status'] == JobStatus.QUEUED
+            raise JobError('Unrecognized answer from server: \n{}'
+                           .format(pprint.pformat(api_job)))
 
-    @property
-    def running(self):
-        """
-        Returns whether job is actively running
-
-        Returns:
-            bool: True if job is running, else False.
+        return self._status
 
-        Raises:
-            QISKitError: couldn't get job status from server
-        """
-        return self.status['status'] == JobStatus.RUNNING
+    def error_message(self):
+        """Returns the error message returned from the API server response"""
+        return self._api_error_msg
 
-    @property
-    def validating(self):
-        """
-        Returns whether job is being validated
+    def queue_position(self):
+        """Returns the position in the server queue
 
         Returns:
-            bool: True if job is under validation, else False.
-
-        Raises:
-            QISKitError: couldn't get job status from server
+            Number: Position in the queue. 0 = No queued yet
         """
-        return self.status['status'] == JobStatus.VALIDATING
+        return self._queue_position
 
-    @property
-    def done(self):
+    def creation_date(self):
         """
-        Returns True if job successfully finished running.
-
-        Note behavior is slightly different than Future objects which would
-        also return true if successfully cancelled.
+        Returns creation date
         """
-        return self.status['status'] == JobStatus.DONE
-
-    @property
-    def cancelled(self):
-        return self._cancelled
+        return self._creation_date
 
-    @property
-    def exception(self):
-        """
-        Return Exception object previously raised by job else None
-
-        Returns:
-            Exception: exception raised by job
-        """
-        if isinstance(self._exception, Exception):
-            self._status_msg = str(self._exception)
-        return self._exception
-
-    @property
+    # pylint: disable=invalid-name
     def id(self):
+        """Return backend determined id.
+
+        If the ID is not set because the job is already initializing, this call will
+        block until we have an ID.
         """
-        Return backend determined id (also available in status method).
-        """
-        # pylint: disable=invalid-name
-        while self._id is None and self._status not in JOB_FINAL_STATES:
-            if self._future_submit.exception():
-                self._status = JobStatus.ERROR
-                self._exception = self._future_submit.exception()
-            # job is initializing and hasn't gotten a id yet.
-            time.sleep(0.1)
+        self._wait_for_submission()
```

Comment:
I guess because you know that an Id is something retrieved from a response of our servers... :) but maybe it's not that clear for a newcomer to the project.